addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/thermostat/NATherm1Handler.java

```diff
@@ -143,13 +149,22 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                         break;
                     }
                     case CHANNEL_SETPOINT_TEMP: {
-                        // Switch the thermostat to manual mode on the desired setpoint for given duration
-                        Calendar cal = Calendar.getInstance();
-                        cal.add(Calendar.MINUTE, getSetPointDefaultDuration());
-                        getBridgeHandler().getThermostatApi().setthermpoint(getParentId(), getId(), "manual",
-                                (int) (cal.getTimeInMillis() / 1000), Float.parseFloat(command.toString()));
-                        updateState(channelUID, new DecimalType(command.toString()));
-                        requestParentRefresh();
+                        BigDecimal spTemp;
+                        if (command instanceof QuantityType) {
+                            spTemp = ((QuantityType<Temperature>) command).toUnit(API_TEMPERATURE_UNIT)
+                                    .toBigDecimal().setScale(1, RoundingMode.HALF_UP);
+                        } else {
+                            spTemp = new BigDecimal(command.toString()).setScale(1, RoundingMode.HALF_UP);
+                        }
+                        if (spTemp != null) {
```

Comment:
It currently looks like a redundant null check. The `QuantityType.toUnit(..)` call could return null according to my compiler. So if you add an additional null check there, this check will no longer be redundant.

Reply:
Sorry, I don't understand what you expect. As you mentioned, the compiler tells me that a NULL check is required and so I aaded it.

Reply:
If we are talking about the same compiler warning there is still a null check missing:

![selection_003](https://user-images.githubusercontent.com/12213581/39446012-5158be9e-4cbd-11e8-92bd-3e46bcb5a862.png)

Currently a NPE occurs before it can even do the null check.

You can solve that with a case that looks like: 

```java
case CHANNEL_SETPOINT_TEMP: {
    BigDecimal spTemp = null;
    if (command instanceof QuantityType) {
        QuantityType<Temperature> quantity = ((QuantityType<Temperature>) command)
                .toUnit(API_TEMPERATURE_UNIT);
        if (quantity != null) {
            spTemp = quantity.toBigDecimal().setScale(1, RoundingMode.HALF_UP);
        }
    } else {
        spTemp = new BigDecimal(command.toString()).setScale(1, RoundingMode.HALF_UP);
    }
    if (spTemp != null) {
        // Switch the thermostat to manual mode on the desired setpoint for given duration
        Calendar cal = Calendar.getInstance();
        cal.add(Calendar.MINUTE, getSetPointDefaultDuration());
        getBridgeHandler().getThermostatApi().setthermpoint(getParentId(), getId(), "manual",
                (int) (cal.getTimeInMillis() / 1000), spTemp.floatValue());
        updateState(channelUID, new QuantityType<Temperature>(spTemp, API_TEMPERATURE_UNIT));
        requestParentRefresh();
    }
    break;
}
```